contrib/cirrus/setup_environment.sh

```diff
@@ -212,6 +212,33 @@ if [[ -n "$ROOTLESS_USER" ]]; then
     echo "ROOTLESS_UID=$ROOTLESS_UID" >> /etc/ci_environment
 fi
 
+# FIXME! experimental workaround for #16973, the "lookup cdn03.quay.io" flake.
+if ((CONTAINER==0)); then
+    nsswitch=/etc/authselect/nsswitch.conf
+    if [[ -e $nsswitch ]]; then
+        if grep -q -E 'hosts:.*resolve' $nsswitch; then
+            msg "Disabling systemd-resolved"
+            sed -i -e 's/^\(hosts: *\).*/\1files dns myhostname/' $nsswitch
+            systemctl stop systemd-resolved
+            rm -f /etc/resolv.conf
+            systemctl start NetworkManager
+
+            # Sigh. The first time, NetworkManager doesn't seem
+            # to create a resolv.conf. Keep retrying until it does.
```

Comment:
You need to restart it, systemd will ignore start when a unit it is already running and AFAIK NetworkManager should already be running unless we explicitly disabled it use.

Reply:
Hm, could be. On a 1minutetip  VM it was not started by default. Anyhow, no flakes in two runs, I'm going to keep hammering away at CI with this to see how it goes. Thanks for the suggestion.